File: NBMBC_iOS/NBMBC_iOS/Models/DataStore.swift
import Foundation
import Combine

@MainActor
class DataStore: ObservableObject {
    @Published var sermons: [Sermon] = []
    @Published var events: [Event] = []
    @Published var groups: [Group] = []
    @Published var isLoading = false

    let church = ChurchInfo()

    init() {
        loadMockData()
    }

    // MARK: - Mock Data (replace with API calls)
    private func loadMockData() {
        let cal = Calendar.current
        let now = Date()

        sermons = [
            Sermon(id: 1, title: "Walking in Purpose", speaker: "Pastor",
                   series: "The Abundant Life", date: cal.date(byAdding: .day, value: -7, to: now)!,
                   description: "Discovering God's unique calling for your life and walking boldly in His purpose for you.",
                   videoURL: "https://www.youtube.com/watch?v=dQw4w9WgXcQ",
                   thumbnailSeed: "sermon1"),
            Sermon(id: 2, title: "Faith Over Fear", speaker: "Pastor",
                   series: "The Abundant Life", date: cal.date(byAdding: .day, value: -14, to: now)!,
                   description: "How to stand firm in faith when the storms of life are raging all around you.",
                   videoURL: nil, thumbnailSeed: "sermon2"),
            Sermon(id: 3, title: "The Power of Prayer", speaker: "Pastor",
                   series: "Prayer Warriors", date: cal.date(byAdding: .day, value: -21, to: now)!,
                   description: "Unlocking the transformative power of a consistent, fervent prayer life.",
                   videoURL: nil, thumbnailSeed: "sermon3"),
            Sermon(id: 4, title: "Grace That Amazes", speaker: "Minister Johnson",
                   series: "Amazing Grace", date: cal.date(byAdding: .day, value: -28, to: now)!,
                   description: "An exploration of God's unmerited favor and how it shapes our daily lives.",
                   videoURL: nil, thumbnailSeed: "sermon4"),
            Sermon(id: 5, title: "Community and Connection", speaker: "Pastor",
                   series: "Building Together", date: cal.date(byAdding: .day, value: -35, to: now)!,
                   description: "Why the local church community is essential to every believer's spiritual growth.",
                   videoURL: nil, thumbnailSeed: "sermon5"),
            Sermon(id: 6, title: "The Prodigal Son", speaker: "Minister Williams",
                   series: "Parables of Jesus", date: cal.date(byAdding: .day, value: -42, to: now)!,
                   description: "A deep dive into one of Jesus' most beloved parables about redemption and the Father's love.",
                   videoURL: nil, thumbnailSeed: "sermon6"),
        ]

        events = [
            Event(id: 1, title: "Sunday Morning Worship", date: nextSunday(from: now), endDate: nil,
                  location: "Main Sanctuary", description: "Join us for our weekly Sunday morning worship service. All are welcome!",
                  imageURL: nil, capacity: nil, icon: "music.note.house.fill"),
            Event(id: 2, title: "Men's Bible Study", date: cal.date(byAdding: .day, value: 3, to: now)!,
                  endDate: nil, location: "Fellowship Hall", description: "A powerful time in the Word for men of all ages.",
                  imageURL: nil, capacity: 40, icon: "book.fill"),
            Event(id: 3, title: "City Wide Mission Outreach", date: cal.date(byAdding: .day, value: 5, to: now)!,
                  endDate: nil, location: "Downtown Winter Haven", description: "Reaching the streets of Winter Haven with love, food, and the Gospel.",
                  imageURL: nil, capacity: nil, icon: "heart.fill"),
            Event(id: 4, title: "Youth Night", date: cal.date(byAdding: .day, value: 8, to: now)!,
                  endDate: nil, location: "Youth Center", description: "A fun and faith-filled evening for teens and young adults.",
                  imageURL: nil, capacity: 75, icon: "star.fill"),
            Event(id: 5, title: "Women's Tea & Fellowship", date: cal.date(byAdding: .day, value: 12, to: now)!,
                  endDate: nil, location: "Fellowship Hall", description: "Ladies, come enjoy fellowship, tea, and uplifting conversation.",
                  imageURL: nil, capacity: 50, icon: "cup.and.heat.waves.fill"),
            Event(id: 6, title: "Annual Church Picnic", date: cal.date(byAdding: .day, value: 18, to: now)!,
                  endDate: nil, location: "Cypress Gardens Park", description: "Bring the whole family for a day of food, fun, and fellowship!",
                  imageURL: nil, capacity: nil, icon: "tree.fill"),
        ]

        groups = [
            Group(id: 1, title: "Kids Ministry", category: "Children",
                  description: "A vibrant, safe place where children discover Jesus and grow in faith through exciting lessons, worship, and activities.",
                  icon: "figure.and.child.holdinghands", leader: "Sis. Davis", meetingTime: "Sundays at 10:30 AM", imageName: nil),
            Group(id: 2, title: "Men's Ministry", category: "Men",
                  description: "Men growing in faith, brotherhood, and servant leadership together. Iron sharpening iron.",
                  icon: "person.3.fill", leader: "Bro. Thompson", meetingTime: "Thursdays at 7 PM", imageName: nil),
            Group(id: 3, title: "Women's Ministry", category: "Women",
                  description: "A sisterhood of faith where women are encouraged, equipped, and empowered to live out their God-given purpose.",
                  icon: "person.2.fill", leader: "Sis. Williams", meetingTime: "Tuesdays at 6:30 PM", imageName: nil),
            Group(id: 4, title: "Youth Ministry", category: "Youth",
                  description: "Raising up the next generation of bold, faith-filled leaders through discipleship, fun, and real community.",
                  icon: "figure.2.arms.open", leader: "Min. Jackson", meetingTime: "Fridays at 7 PM", imageName: nil),
            Group(id: 5, title: "City Wide Mission", category: "Outreach",
                  description: "Reaching the streets of Winter Haven with love, food, and the Gospel. Serving our city in the name of Jesus.",
                  icon: "heart.circle.fill", leader: "Deacon Brown", meetingTime: "1st Saturdays at 9 AM", imageName: nil),
            Group(id: 6, title: "Senior Saints", category: "Seniors",
                  description: "Honoring and celebrating our faithful seniors through fellowship, Bible study, and shared wisdom.",
                  icon: "person.crop.circle.badge.checkmark", leader: "Min. Roberts", meetingTime: "Wednesdays at 11 AM", imageName: nil),
        ]
    }

    private func nextSunday(from date: Date) -> Date {
        var cal = Calendar.current
        cal.firstWeekday = 1
        var comps = cal.dateComponents([.weekday], from: date)
        let daysUntilSunday = (8 - (comps.weekday ?? 1)) % 7
        let offset = daysUntilSunday == 0 ? 7 : daysUntilSunday
        var target = cal.date(byAdding: .day, value: offset, to: date)!
        comps = cal.dateComponents([.year, .month, .day], from: target)
        comps.hour = 10
        comps.minute = 30
        comps.second = 0
        return cal.date(from: comps) ?? target
    }

    var latestSermons: [Sermon] { Array(sermons.prefix(3)) }
    var upcomingEvents: [Event] { Array(events.prefix(3)) }
}

File: NBMBC_Android/app/src/main/java/com/rcdev/nbmbc/data/DataStore.kt
package com.rcdev.nbmbc.data

import androidx.lifecycle.ViewModel
import kotlinx.coroutines.flow.MutableStateFlow
import kotlinx.coroutines.flow.asStateFlow
import java.time.LocalDate
import java.time.LocalDateTime

class AppViewModel : ViewModel() {
    val church = ChurchInfo()

    private val _sermons = MutableStateFlow(mockSermons())
    val sermons = _sermons.asStateFlow()

    private val _events = MutableStateFlow(mockEvents())
    val events = _events.asStateFlow()

    private val _groups = MutableStateFlow(mockGroups())
    val groups = _groups.asStateFlow()

    val latestSermons get() = _sermons.value.take(3)
    val upcomingEvents get() = _events.value.take(3)
}

private fun mockSermons(): List<Sermon> {
    val today = LocalDate.now()
    return listOf(
        Sermon(1, "Walking in Purpose", "Pastor", "The Abundant Life", today.minusDays(7),
            "Discovering God's unique calling for your life and walking boldly in His purpose.", "https://www.youtube.com/watch?v=dQw4w9WgXcQ"),
        Sermon(2, "Faith Over Fear", "Pastor", "The Abundant Life", today.minusDays(14),
            "How to stand firm in faith when the storms of life are raging around you."),
        Sermon(3, "The Power of Prayer", "Pastor", "Prayer Warriors", today.minusDays(21),
            "Unlocking the transformative power of a consistent, fervent prayer life."),
        Sermon(4, "Grace That Amazes", "Minister Johnson", "Amazing Grace", today.minusDays(28),
            "An exploration of God's unmerited favor and how it shapes our daily lives."),
        Sermon(5, "Community and Connection", "Pastor", "Building Together", today.minusDays(35),
            "Why the local church community is essential to every believer's spiritual growth."),
        Sermon(6, "The Prodigal Son", "Minister Williams", "Parables of Jesus", today.minusDays(42),
            "A deep dive into one of Jesus' most beloved parables about redemption."),
    )
}

private fun mockEvents(): List<Event> {
    val now = LocalDateTime.now()
    return listOf(
        Event(1, "Sunday Morning Worship", now.plusDays(6).withHour(10).withMinute(30),
            "Main Sanctuary", "Join us for our weekly Sunday morning worship service. All are welcome!", null, "church"),
        Event(2, "Men's Bible Study", now.plusDays(3).withHour(19).withMinute(0),
            "Fellowship Hall", "A powerful time in the Word for men of all ages.", 40, "book"),
        Event(3, "City Wide Mission Outreach", now.plusDays(5).withHour(9).withMinute(0),
            "Downtown Winter Haven", "Reaching the streets of Winter Haven with love, food, and the Gospel.", null, "favorite"),
        Event(4, "Youth Night", now.plusDays(8).withHour(19).withMinute(0),
            "Youth Center", "A fun and faith-filled evening for teens and young adults.", 75, "star"),
        Event(5, "Women's Tea & Fellowship", now.plusDays(12).withHour(14).withMinute(0),
            "Fellowship Hall", "Ladies, come enjoy fellowship, tea, and uplifting conversation.", 50, "local_cafe"),
        Event(6, "Annual Church Picnic", now.plusDays(18).withHour(11).withMinute(0),
            "Cypress Gardens Park", "Bring the whole family for a day of food, fun, and fellowship!", null, "park"),
    )
}

private fun mockGroups(): List<Group> = listOf(
    Group(1, "Kids Ministry", "Children", "A vibrant, safe place where children discover Jesus and grow in faith through exciting lessons, worship, and activities.", "child_care", "Sis. Davis", "Sundays at 10:30 AM"),
    Group(2, "Men's Ministry", "Men", "Men growing in faith, brotherhood, and servant leadership together. Iron sharpening iron.", "group", "Bro. Thompson", "Thursdays at 7 PM"),
    Group(3, "Women's Ministry", "Women", "A sisterhood of faith where women are encouraged, equipped, and empowered to live out their God-given purpose.", "people", "Sis. Williams", "Tuesdays at 6:30 PM"),
    Group(4, "Youth Ministry", "Youth", "Raising up the next generation of bold, faith-filled leaders through discipleship, fun, and real community.", "sports_esports", "Min. Jackson", "Fridays at 7 PM"),
    Group(5, "City Wide Mission", "Outreach", "Reaching the streets of Winter Haven with love, food, and the Gospel. Serving our city in the name of Jesus.", "favorite", "Deacon Brown", "1st Saturdays at 9 AM"),
    Group(6, "Senior Saints", "Seniors", "Honoring and celebrating our faithful seniors through fellowship, Bible study, and shared wisdom.", "elderly", "Min. Roberts", "Wednesdays at 11 AM"),
)
